### ISSUE TITLE
Gets stuck if does not get a server response

### ISSUE THREAD

**Issue by tiberiusferreira** [voider1/teleborg#21]
Hello, I've been using this lib and it's great. I had for a long time a problem when it stopped responding after some days. 

After some digging I think the source of the problem is that the reqwest request does not have a timeout. So if it doesn't hear back from the server it just hangs forever. 

To do a quick fix I had to update reqwest to a newer version which has an option to add a timeout, as per https://github.com/tiberiusferreira/teleborg/commit/bcd4a541ec6d9a1e2def00b664855363e9640ecf .

Anyway, just opening this issue because some other users might run into this problem.

**Comment by voider1** [comment]
Yeah, I've noticed this problem myself. I have very little time to work on it right now because I'm kind of busy nowadays. I do intend to come back to it someday. Thanks for the feedback!
